spec/rspec/rails/matchers/have_reported_error_spec.rb

```diff
@@ -0,0 +1,233 @@
+require "rspec/rails/matchers/have_reported_error"
+
+RSpec.describe "have_reported_error matcher" do
+  class TestError < StandardError; end
+  class AnotherTestError < StandardError; end
+
+  let(:mock_error_reporter) { double("ErrorReporter") }
+  let(:subscribers) { [] }
+
+  before do
+    allow(Rails).to receive(:error).and_return(mock_error_reporter)
+
+    allow(mock_error_reporter).to receive(:subscribe) do |subscriber|
+      subscribers << subscriber
+    end
+
+    allow(mock_error_reporter).to receive(:unsubscribe) do |subscriber|
+      subscribers.delete(subscriber)
+    end
+
+    allow(mock_error_reporter).to receive(:report) do |error, **attrs|
+      subscribers.each { |subscriber| subscriber.report(error, **attrs) }
+    end
+  end
+
+  after do
+    subscribers.clear
```

Comment:
This is unlikely to be needed, as ‘let’ will initialize a bew array each time, won’t it?